### PR TITLE
[css-shapes] Comprehensive tests for clip-path: shape()

### PR DESCRIPTION
**Pull request by noamr**

### CODE REVIEW

**Comment by birtles** [comment]
Hi @BorisChiou -- sorry to throw so many tasks your way at the same time. I added you as a reviewer of this because I think you did the shape interpolation in Gecko. Is that right?

**Comment by BorisChiou** [comment]
I did the CSS `path()` interpolation and updated some code for other basic-shape functions in Gecko. Looks like this is a new function, `shape()` from https://drafts.csswg.org/css-shapes-2/#funcdef-shape. Sure, I can take a look at these tests. :)

**Comment by noamr** [comment]
> I did the CSS `path()` interpolation and updated some code for other basic-shape functions in Gecko. Looks like this is a new function, `shape()` from https://drafts.csswg.org/css-shapes-2/#funcdef-shape. Sure, I can take a look at these tests. :)

I contibuted the `shape()` definition and working on some draft (webkit) implementation which allowed me to write tests that work :)
Wanted to contribute the tests before working on upstreaming the implementation

File: css/css-masking/clip-path/reference/clip-path-shape-003-ref.html
Comment (BorisChiou) [inline css/css-masking/clip-path/reference/clip-path-shape-003-ref.html:7]: nit: we don't need to add meta "assert" for reference file

File: css/css-masking/clip-path/reference/clip-path-shape-003-ref.html
Comment (BorisChiou) [inline css/css-masking/clip-path/reference/clip-path-shape-003-ref.html:4]: nit: Perhaps using "Reference of xxx" would be better.

File: css/css-masking/clip-path/reference/clip-path-shape-004-ref.html
Comment (BorisChiou) [inline css/css-masking/clip-path/reference/clip-path-shape-004-ref.html:4]: nit: Perhaps using "Reference of xxx" would be better.

File: css/css-masking/clip-path/reference/clip-path-shape-004-ref.html
Comment (BorisChiou) [inline css/css-masking/clip-path/reference/clip-path-shape-004-ref.html:7]: nit: We don't need to add meta "assert" for reference file

File: css/css-masking/clip-path/clip-path-shape-002-units.html
Comment (BorisChiou) [inline css/css-masking/clip-path/clip-path-shape-002-units.html:4]: s/nonzero/evenodd/

File: css/css-masking/clip-path/clip-path-shape-002-units.html
Comment (BorisChiou) [inline css/css-masking/clip-path/clip-path-shape-002-units.html:9]: s/nonzero/evenodd/

File: css/css-masking/clip-path/clip-path-shape-interpolation-002.html
Comment (BorisChiou) [inline css/css-masking/clip-path/clip-path-shape-interpolation-002.html:4]: s/nonzero/evenodd/

File: css/css-masking/clip-path/clip-path-shape-interpolation-002.html
Comment (BorisChiou) [inline css/css-masking/clip-path/clip-path-shape-interpolation-002.html:8]: s/nonzero/evenodd/

File: css/css-masking/animations/clip-path-interpolation-shape.html
Comment (BorisChiou) [inline css/css-masking/animations/clip-path-interpolation-shape.html:142]: Is this duplicated? Looks like this is the same as the previous one.

File: css/css-masking/animations/clip-path-interpolation-shape.html
Comment (BorisChiou) [inline css/css-masking/animations/clip-path-interpolation-shape.html:120]: Is this duplicated? Looks like this is the same as the previous one.

**Review by BorisChiou — CHANGES_REQUESTED** [review]
Overall looks good, but have some typos and nits. Would you mind updating them before merging? I request changes for now and will approve them after you update these nits. Thanks.

Basically, `clip-path-shape-interpolation-*.html` should be in the animation folder because these two files test the css animations. However, this is my bad because I added `clip-path-path-interpolation-*.html` in the same places. This patch looks good, and perhaps we should move them into animation folder in the future.

File: css/css-masking/animations/clip-path-interpolation-shape.html
Comment (noamr) [reply css/css-masking/animations/clip-path-interpolation-shape.html:120]: It uses `by` instead of `to`

File: css/css-masking/animations/clip-path-interpolation-shape.html
Comment (noamr) [reply css/css-masking/animations/clip-path-interpolation-shape.html:142]: by instead of to

File: css/css-masking/parsing/clip-path-shape-parsing.html
Comment (noamr) [reply css/css-masking/parsing/clip-path-shape-parsing.html]: Fixed

File: css/css-masking/parsing/clip-path-shape-parsing.html
Comment (noamr) [reply css/css-masking/parsing/clip-path-shape-parsing.html]: Fixed

File: css/css-masking/clip-path/reference/clip-path-shape-004-ref.html
Comment (noamr) [reply css/css-masking/clip-path/reference/clip-path-shape-004-ref.html]: Fixed

File: css/css-masking/clip-path/reference/clip-path-shape-004-ref.html
Comment (noamr) [reply css/css-masking/clip-path/reference/clip-path-shape-004-ref.html]: Fixed

File: css/css-masking/clip-path/reference/clip-path-shape-003-ref.html
Comment (noamr) [reply css/css-masking/clip-path/reference/clip-path-shape-003-ref.html]: Fixed

File: css/css-masking/clip-path/reference/clip-path-shape-003-ref.html
Comment (noamr) [reply css/css-masking/clip-path/reference/clip-path-shape-003-ref.html]: Fixed

File: css/css-masking/clip-path/clip-path-shape-interpolation-002.html
Comment (noamr) [reply css/css-masking/clip-path/clip-path-shape-interpolation-002.html]: Fixed

File: css/css-masking/clip-path/clip-path-shape-interpolation-002.html
Comment (noamr) [reply css/css-masking/clip-path/clip-path-shape-interpolation-002.html]: Fixed

File: css/css-masking/clip-path/clip-path-shape-002.html
Comment (noamr) [reply css/css-masking/clip-path/clip-path-shape-002.html]: Fixed

File: css/css-masking/clip-path/clip-path-shape-002-units.html
Comment (noamr) [reply css/css-masking/clip-path/clip-path-shape-002-units.html]: Fixed

File: css/css-masking/clip-path/clip-path-shape-002-units.html
Comment (noamr) [reply css/css-masking/clip-path/clip-path-shape-002-units.html]: Fixed

**Review by BorisChiou — APPROVED** [review]
Thanks!

**Pull request merged**
